<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Widget</class>
 <widget class="QWidget" name="Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1770</width>
    <height>800</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1770</width>
    <height>800</height>
   </size>
  </property>
  <property name="font">
   <font>
    <family>微软雅黑 Light</family>
   </font>
  </property>
  <property name="windowTitle">
   <string>VisualSorting @LuiYK</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_3">
   <item>
    <layout class="QHBoxLayout" name="MainWidget">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_Left">
       <item>
        <widget class="QComboBox" name="comboBox_sorting">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="font">
          <font>
           <family>微软雅黑 Light</family>
           <pointsize>10</pointsize>
          </font>
         </property>
         <property name="contextMenuPolicy">
          <enum>Qt::DefaultContextMenu</enum>
         </property>
         <item>
          <property name="text">
           <string>Bubble Sort</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Bubble Sort(Recursive)</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Cocktail Sort</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Gnome Sort</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Selection Sort</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Insertion Sort</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Quick Sort</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Quick Sort(2ways)</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Quick Sort(3ways)</string>
          </property>
         </item>
        </widget>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QLabel" name="label">
             <property name="font">
              <font>
               <family>微软雅黑 Light</family>
               <pointsize>10</pointsize>
              </font>
             </property>
             <property name="text">
              <string>Array size(*10):</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="spinBox_ArraySize">
             <property name="minimumSize">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
             <property name="font">
              <font>
               <family>微软雅黑 Light</family>
               <pointsize>10</pointsize>
              </font>
             </property>
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>149</number>
             </property>
             <property name="singleStep">
              <number>2</number>
             </property>
             <property name="value">
              <number>1</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="Line" name="line_2">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_Comprision_2">
           <property name="font">
            <font>
             <family>微软雅黑 Light</family>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Comprision Delay(ms):</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSlider" name="horizontalSlider">
           <property name="minimum">
            <number>1</number>
           </property>
           <property name="maximum">
            <number>200</number>
           </property>
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="invertedAppearance">
            <bool>false</bool>
           </property>
           <property name="invertedControls">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="Line" name="line">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_A">
         <item>
          <widget class="QLabel" name="label_ArrayAccess">
           <property name="font">
            <font>
             <family>微软雅黑 Light</family>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Array Access:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_AccessCount">
           <property name="font">
            <font>
             <family>微软雅黑 Light</family>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Access</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_C">
         <item>
          <widget class="QLabel" name="label_Comprision">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="font">
            <font>
             <family>微软雅黑 Light</family>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="text">
            <string>Comprision Count:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLabel" name="label_CpCount">
           <property name="sizePolicy">
            <sizepolicy hsizetype="MinimumExpanding" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="font">
            <font>
             <family>微软雅黑 Light</family>
             <pointsize>10</pointsize>
            </font>
           </property>
           <property name="text">
            <string>CpC123</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::MinimumExpanding</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <layout class="QVBoxLayout" name="verticalLayout"/>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_ramdonArray">
         <property name="font">
          <font>
           <family>微软雅黑 Light</family>
           <pointsize>10</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Random Array</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pushButton_sort">
         <property name="font">
          <font>
           <family>微软雅黑 Light</family>
           <pointsize>10</pointsize>
          </font>
         </property>
         <property name="text">
          <string>Sort</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_2"/>
     </item>
     <item>
      <widget class="QGraphicsView" name="graphicsView_visualSorting">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="minimumSize">
        <size>
         <width>1500</width>
         <height>750</height>
        </size>
       </property>
       <property name="maximumSize">
        <size>
         <width>1500</width>
         <height>750</height>
        </size>
       </property>
       <property name="font">
        <font>
         <family>微软雅黑 Light</family>
        </font>
       </property>
       <property name="styleSheet">
        <string notr="true">
background-color:rgb(0,0,0)</string>
       </property>
       <property name="frameShape">
        <enum>QFrame::NoFrame</enum>
       </property>
       <property name="verticalScrollBarPolicy">
        <enum>Qt::ScrollBarAlwaysOff</enum>
       </property>
       <property name="horizontalScrollBarPolicy">
        <enum>Qt::ScrollBarAlwaysOff</enum>
       </property>
       <property name="renderHints">
        <set>QPainter::Antialiasing|QPainter::TextAntialiasing</set>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
